python/cuml/dask/preprocessing/encoders.py

```diff
@@ -0,0 +1,357 @@
+                             'are supported when using sparse')
+
+    def _validate_keywords(self):
+        if self.handle_unknown not in ('error', 'ignore'):
+            msg = ("handle_unknown should be either 'error' or 'ignore', "
+                   "got {0}.".format(self.handle_unknown))
+            raise ValueError(msg)
+        # If we have both dropped columns and ignored unknown
+        # values, there will be ambiguous cells. This creates difficulties
+        # in interpreting the model.
+        if self.drop is not None and self.handle_unknown != 'error':
+            raise ValueError(
+                "`handle_unknown` must be 'error' when the drop parameter is "
+                "specified, as both would create categories that are all "
+                "zero.")
+
+    def _check_is_fitted(self):
+        if not self._fitted:
+            raise RuntimeError("Model must first be .fit()")
+
+    def _compute_drop_idx(self):
+        if self.drop is None:
+            return None
+        elif isinstance(self.drop, str) and self.drop == 'first':
+            return {feature: cp.array([0])
+                    for feature in self._encoders.keys()}
+        elif isinstance(self.drop, dict):
+            if len(self.drop.keys()) != len(self._encoders):
+                msg = ("`drop` should have as many columns as the number "
+                       "of features ({}), got {}")
+                raise ValueError(msg.format(len(self._encoders),
+                                            len(self.drop.keys())))
+            drop_idx = dict()
+            for feature in self.drop.keys():
+                cats = self._encoders[feature]
+                self.drop[feature] = Series(self.drop[feature])
+                if len(self.drop[feature]) != 1:
+                    msg = ("Trying to drop multiple values for feature {}, "
+                           "this is not supported.").format(feature)
+                    raise ValueError(msg)
+                if not self.drop[feature].isin(cats).all():
+                    msg = ("Some categories for feature {} were supposed "
+                           "to be dropped, but were not found in the encoder "
+                           "categories.".format(feature))
+                    raise ValueError(msg)
+                idx = cats.isin(self.drop[feature])
+                drop_idx[feature] = cp.asarray(cats[idx].index)
+            return drop_idx
+        else:
+            msg = ("Wrong input for parameter `drop`. Expected "
+                   "'first', None or a dict, got {}")
+            raise ValueError(msg.format(type(self.drop)))
+
+    def get_categories_(self):
+        """
+        Returns categories used for the one hot encoding in the correct order.
+
+        This copies the categories to the CPU and should only be used to check
+        the order of the categories.
+        """
+        return [self._encoders[f].to_array() for f in self._features]
```

Comment:
If this is going to be a public method, we should keep the categories on device and provide better pydocs for the inputs and outputs. I don't see this method on sklearn's API, so another option could just be to make it internal and prefix it with `_`.

Reply:
Sklearn has `categories_` which is used to check the categories of each feature during fitting (in order of the features in X and corresponding with the output of `transform`). In order to have the same capability I added this function and named it like a function since it's not really an attribute like Sklearn's estimator. I could rename it to `categories_` if it makes more sense.

Reply:
It's not used internally, it is purely an helper for the user.